qp_klp/Amplicon.py

```diff
@@ -81,23 +86,59 @@ def quality_control(self):
                 new_path = join(output_folder, basename(raw_fastq_file))
                 copyfile(raw_fastq_file, new_path)
 
-    def generate_reports(self, input_file_path):
-        super()._generate_reports(self.pipeline.mapping_file)
+    def generate_reports(self):
+        super()._generate_reports()
         return None  # amplicon doesn't need project names
 
+    def _get_data_type(self, prep_file_path):
+        metadata = Step.parse_prep_file(prep_file_path)
+        if 'target_gene' in metadata.columns:
+            tg = set(metadata['sample target_gene'])
+            if len(tg) != 1:
+                raise ValueError("More than one value for target_gene")
+            else:
```

Comment:
Do you really need the else here?

Reply:
IMHO, sanity checks can be important. In this case there are plenty of legacy prep-info-files out there w/out target_gene as a column. Granted, they might not pass through this code but I think it's best to raise the Error if a value can't be found, rather than continue and output an unpredictable file.

Reply:
Right, note that my question wasn't the if but the else ... I do not think is necessary.